lib/msprime.c

```diff
@@ -3407,6 +4071,24 @@ msp_reset_memory_state(msp_t *self)
                 avl_unlink_node(&pop->ancestors[label], node);
                 msp_free_avl_node(self, node);
             }
+            if (pop->hulls_left != NULL) {
```

Comment:
This case isn't being hit in tests - I think you need to run some replicates of the smc_k?

Reply:
This is due to the fact that when we reset the state we reset to 'hudson' (= initial_model) first before setting the model to smc_k. This is a remainder of the code from when before I realised that when going through the python API we always reset to 'hudson' first.